### DISCUSSION TITLE
How to enable storage in memory to enable compression?

### DISCUSSION THREAD

**Discussion by candux** [duckdb/duckdb-wasm#1470] [Q&A] [answered]
I want to import a big table from CSV which doesn't fit into the 4GB Ram limit of wasm. 
If I import it into DuckDB-CLI with storage enabled, [the compression](https://duckdb.org/2022/10/28/lightweight-compression.html) reduces the size and the used Ram a lot. To have compression in duckdb-wasm, I think I need to enable the storage on the Virtual Filesystem. Is this possible? Can somebody please give me a hint how to do this?

**Comment by candux** [comment, marked as answer]
I found the solution. Just create a new database and use it :)
```
ATTACH DATABASE 'duckdb' as mydb;
use mydb;
```
Now all new tables will use compression. This works great and I can now import much bigger tables. The Memory-Consumption of a very sparse table went down from 2.3GB to only about 400MB (memory + database size)!

**Reply by thrmagic** [reply]
How to compression,can you give me a demo，thanks.

**Reply by nshiab** [reply]
Unfortunately, this doesn't work anymore with @1.28.1-dev258.0. Do you have another solution @candux ?

<img width="563" alt="Screenshot 2024-09-09 at 9 04 13 PM" src="https://github.com/user-attachments/assets/fe623291-1e27-4400-aa57-f44153f1fac8">

**Comment by nshiab** [comment]
I don't understand why, but this works great! I was stuck with operations on a 1go CSV file, which were eating up all the memory. These two lines fixed the problem!

**Reply by thrmagic** [reply]
Where should I put these two lines of code，thanks

**Reply by nshiab** [reply]
> Where should I put these two lines of code，thanks

At the beginning of your query.

**Reply by thrmagic** [reply]
thanks,But,the javascript can’t identify these two lines of code,Do you know how to compress to reduce size.
